tap_bitbucket/gitlocal.py

```diff
@@ -53,10 +56,18 @@ def hashPatchLine(patchLine, hmacToken = None):
 
 
 def parseDiffLines(lines, shouldEncrypt=False, hmacToken=None):
+  MAX_DIFF_LINES=1000
+  MAX_PATCH_SIZE=1024 * 1024
+  MAX_LINE_LENGTH=500
+
   changes = []
   curChange = None
   state = 'start'
   for line in lines:
+    # Replace any invalid characters
+    # Also, don't allow nulls since we are treating data as strings downstream. (FFFD is unicode
+    # replacement character.)
+    line = line[0:MAX_LINE_LENGTH-1].replace('\u0000', '\uFFFD')
```

Comment:
Now that we are streaming lines, we still need to verify that we don't have null unicode characters.

Reply:
If we encounter this we should count the file as binary and stop saving the diff. I'm a bit surprised this is happening with git diff though because it should already not be showing binary files, but maybe it uses different criteria to make that binary determination.